daemon/cmd/kube_proxy_replacement.go

```diff
@@ -307,6 +307,28 @@ func detectNativeDevices(strict bool) {
 // replacement after all devices are known.
 func finishKubeProxyReplacementInit(isKubeProxyReplacementStrict bool) {
 	if option.Config.EnableNodePort {
+
+		// Check whether the DirectRoutingDevice (if specified) is defined within devices
+		if option.Config.DirectRoutingDevice != "" {
+			directDev := option.Config.DirectRoutingDevice
+			directDevFound := false
+			for _, iface := range option.Config.Devices {
+				if iface == directDev { // found
+					directDevFound = true
+					break
+				}
+			}
+
+			if !directDevFound {
+				// docs state that:
+				//   If the direct routing device does not exist within
+				//   devices, Cilium will add the device to the latter
+				//   list.
+				log.Infof("DirectRoutingDevice %s not included in specified devices %v: adding it", directDev, option.Config.Devices)
```

Comment:
also I think we probably don't need to emit an extra info message here given we fix this up anyway on behalf of the user?

Reply:
rest lgtm

Reply:
I think the message might be useful if the user specified their own routing devices and we need to add one to the one they specified. But, I'll remove it to avoid confusion.